textinput/textinput.go

```diff
@@ -154,6 +160,15 @@ type Model struct {
 	// cursorMode determines the behavior of the cursor
 	cursorMode CursorMode
 
+	// Should the input suggest to complete
+	ShowCompletions bool
+
+	// Key to be pressed in order to accept the complete suggestion, defaults to right-arrow
+	AcceptCompletionKey tea.KeyType
```

Comment:
We'll want to be able to cycle through suggestions, so we should add two keys to cycle forward and backwards through them.
```suggestion
	AcceptSuggestionKey tea.KeyType
	NextSuggestionKey   tea.KeyType
    PrevSuggestionKey   tea.KeyType
```